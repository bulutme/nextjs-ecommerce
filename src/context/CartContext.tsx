import React, {
  createContext,
  useContext,
  ReactNode,
  useEffect,
  useState,
} from "react";
import { ProductItemProps } from "@/app/types/Product/types";
import useLocalStorage from "@/hooks/useLocalStorage";

type CartItem = {
  product: ProductItemProps;
  count: number;
};

type GroupedCart = {
  brand: string;
  products: CartItem[];
};

type CartContextType = {
  cart: GroupedCart[];
  addToCart: (product: ProductItemProps) => void;
  removeProduct: (productId: number) => void;
  increaseCount: (productId: number) => void;
  decreaseCount: (productId: number) => void;
  totalItemCount: number;
  totalPrice: string;
  isLoading: boolean;
};

const CartContext = createContext<CartContextType | undefined>(undefined);

const CartProvider = ({ children }: { children: ReactNode }) => {
  const [cart, setCart] = useLocalStorage<GroupedCart[]>("cart", []);
  const [isLoading, setIsLoading] = useState<boolean>(true);

  useEffect(() => {
    if (typeof window !== "undefined") {
      setIsLoading(true);
      setCart(JSON.parse(localStorage.getItem("cart") as string));
      setIsLoading(false);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const totalItemCount = cart.reduce(
    (total, group) =>
      total +
      group.products.reduce((groupTotal, item) => groupTotal + item.count, 0),
    0
  );

  const totalPrice = cart.reduce(
    (total, group) =>
      total +
      group.products.reduce(
        (groupTotal, item) => groupTotal + item.count * item.product.price,
        0
      ),
    0
  );

  const formattedTotalPrice = totalPrice.toFixed(2);

  const addToCart = (product: ProductItemProps) => {
    const existingGroupIndex = cart.findIndex(
      (group) => group.brand === product.brand
    );

    if (existingGroupIndex !== -1) {
      const existingProductIndex = cart[existingGroupIndex].products.findIndex(
        (item) => item.product.id === product.id
      );

      const newCart = [...cart];

      if (existingProductIndex !== -1) {
        const cartId =
          newCart[existingGroupIndex].products[existingProductIndex].product.id;
        increaseCount(cartId);
      } else {
        newCart[existingGroupIndex].products.push({
          product,
          count: 1,
        });
        setCart(newCart);
      }
    } else {
      setCart((prevCart) => [
        ...prevCart,
        {
          brand: product.brand,
          products: [
            {
              product,
              count: 1,
            },
          ],
        },
      ]);
    }
  };

  const removeProduct = (productId: number) => {
    const newCart = cart.map((group) => ({
      ...group,
      products: group.products.filter((item) => item.product.id !== productId),
    }));

    // Remove empty groups
    const updatedCart = newCart.filter((group) => group.products.length > 0);
    setCart(updatedCart);
  };

  const increaseCount = (productId: number) => {
    const newCart = cart.map((group) => {
      return {
        ...group,
        products: group.products.map((item) =>
          item.product.id === productId
            ? { ...item, count: item.count + 1 }
            : item
        ),
      };
    });

    setCart(newCart);
  };

  const decreaseCount = (productId: number) => {
    const newCart = cart.map((group) => ({
      ...group,
      products: group.products.map((item) =>
        item.product.id === productId
          ? { ...item, count: Math.max(1, item.count - 1) }
          : item
      ),
    }));

    setCart(newCart);
  };

  const contextValue: CartContextType = {
    cart,
    totalItemCount,
    totalPrice: formattedTotalPrice,
    addToCart,
    removeProduct,
    increaseCount,
    decreaseCount,
    isLoading,
  };

  return (
    <CartContext.Provider value={contextValue}>{children}</CartContext.Provider>
  );
};

const useCart = () => {
  const context = useContext(CartContext);

  if (!context) {
    throw new Error("useCart must be used within a CartProvider");
  }
  return context;
};

export { CartProvider, useCart };
