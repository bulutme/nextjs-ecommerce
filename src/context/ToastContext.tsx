import { createContext, FC, ReactNode, useContext, useState } from "react";

export type ToastType = "success" | "error" | "exit";

type ToastData = {
  id: number;
  type: ToastType;
  message: string;
};

type ToastContextType = {
  toasts: ToastData[];
  addToast: (type: ToastType, message: string) => void;
  removeToast: (id: number) => void;
};

const MAX_TOASTS = 5;

const ToastContext = createContext<ToastContextType | undefined>(undefined);

export const ToastProvider: FC<{ children: ReactNode }> = ({ children }) => {
  const [toasts, setToasts] = useState<ToastData[]>([]);

  // function to add toast using type and message
  const addToast = (type: ToastType, message: string) => {
    const id = new Date().getTime();
    const newToast: ToastData = { id, type, message };

    // remove the latest one if toast item more than 5
    setToasts((prevToasts) => [...prevToasts.slice(-MAX_TOASTS + 1), newToast]);

    if (type !== "exit") {
      setTimeout(() => {
        removeToast(id);
      }, 3000);
    }
  };

  // function to remove toast
  const removeToast = (id: number) => {
    setToasts((prevToasts) => prevToasts.filter((toast) => toast.id !== id));
  };

  const contextValue: ToastContextType = {
    toasts,
    addToast,
    removeToast,
  };

  return (
    <ToastContext.Provider value={contextValue}>
      {children}
    </ToastContext.Provider>
  );
};

// custom hook to use toast context
export const useToast = () => {
  const context = useContext(ToastContext);
  if (!context) {
    throw new Error("useToast must be used within a ToastProvider");
  }
  return context;
};
