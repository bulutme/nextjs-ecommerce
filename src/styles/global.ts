"use client";

import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 16px;
  }

  html, body, #__next {
    height: 100%;
  }

  a {
    text-decoration: none;
    width: fit-content;
  }

  ul {
    list-style-type: none
  }
`;

export default GlobalStyles;
