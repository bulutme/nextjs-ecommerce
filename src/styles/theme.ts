export const theme = {
  colors: {
    black: "#000",
    white: "#fff",
    gray: "#808080",
    grayLight: "#d3d3d3",
    border: "#e2e8f0",
    primary: "#3bb77e",
    primaryDark: "#2b8a5e",
    error: "#dc2626",
    info: "#0891b2",
    success: "#16a34a",
  },
  screens: {
    "2xs": "400px",
    xs: "480px",
    sm: "576px",
    md: "768px",
    lg: "992px",
    xl: "1200px",
    "2xl": "1440px",
    "3xl": "1560px",
  },
  fontWeight: {
    normal: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
  },
};
