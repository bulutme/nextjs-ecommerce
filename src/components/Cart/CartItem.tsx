import styled, { css } from "styled-components";
import Image from "next/image";
import Link from "next/link";
import { AiOutlineClose } from "react-icons/ai";
import { CartItemProps } from "@/app/types/Product/types";
import { useCart } from "@/context/CartContext";
import { FaMinus, FaPlus } from "react-icons/fa";
import { ToastType, useToast } from "@/context/ToastContext";

const sharedStyles = css`
  display: flex;
  align-items: center;
`;

const CartItemContainer = styled.div`
  background-color: white;
  padding: 15px;
  border: 1px solid ${({ theme }) => theme.colors.border};
  border-radius: 8px;
`;

const CartItemListItem = styled.li`
  position: relative;
  ${sharedStyles}
  padding: 12px 0;
`;

const ProductName = styled.p`
  font-size: 1rem;
  font-weight: ${({ theme }) => theme.fontWeight.medium};
`;

const ProductBrand = styled(ProductName)`
  color: ${({ theme }) => theme.colors.info};
  margin-bottom: 0.6rem;
`;

const ProductImageLink = styled(Link)`
  min-width: fit-content;
`;

const ProductImage = styled(Image)`
  object-fit: cover;
  aspect-ratio: 1 / 1;
  cursor: pointer;
`;

const ProductDetails = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  font-normal;
  font-size: 0.875rem;
  margin-left: 1rem;
`;

const Description = styled.p`
  font-size: 1rem;
  color: rgb(31 41 55);
`;

const QuantityWrapper = styled.div`
  ${sharedStyles}
  gap: 8px;
  padding: 6px 0;
`;

const PriceAndQuantity = styled.div`
  ${sharedStyles}
  justify-content: space-between;
`;

const Quantity = styled.span`
  width: 12px;
  height: 24px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Price = styled.p`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1rem;
  font-weight: ${({ theme }) => theme.fontWeight.semibold};
`;

const CloseIcon = styled(AiOutlineClose)`
  position: absolute;
  width: 28px;
  height: 28px;
  top: 0;
  right: 0;
  padding: 6px;
  border-radius: 50%;
  max-right: 0;
  cursor: pointer;
  transition: opacity 0.3s, transform 0.3s;

  &:hover {
    background: rgb(243 244 246);
  }
`;

const StyledButton = styled.button`
  ${sharedStyles}
  justify-content: center;
  width: 24px;
  height: 24px;
  border-radius: 4px;
  cursor: pointer;
  background-color: ${({ theme }) => theme.colors.primary};
  color: white;
  border: 1px solid ${({ theme }) => theme.colors.primary};
  transition: background-color 0.3s, color 0.3s;

  &:hover {
    background-color: white;
    color: ${({ theme }) => theme.colors.primary};
  }

  &:disabled {
    background-color: ${({ theme }) => theme.colors.grayLight};
    color: ${({ theme }) => theme.colors.white};
    border-color: ${({ theme }) => theme.colors.grayLight};
    cursor: not-allowed;
  }
`;

const CartItem: React.FC<CartItemProps> = ({ brand, products }) => {
  const { removeProduct, increaseCount, decreaseCount } = useCart();
  const { addToast } = useToast();

  const showToast = (type: ToastType, message: string) => {
    addToast(type, message);
  };

  const handleDecreaseCount = (id: number, name: string) => {
    decreaseCount(id);
    showToast("success", `${name}'s quantity decreased`);
  };

  const handleIncreaseCount = (id: number, name: string) => {
    increaseCount(id);
    showToast("success", `${name}'s quantity increased`);
  };

  const handleRemoveProduct = (id: number, name: string) => {
    removeProduct(id);
    showToast("success", `${name} deleted from cart`);
  };

  return (
    <CartItemContainer>
      <ul>
        <Link href="#">
          <ProductBrand>{brand}</ProductBrand>
        </Link>
        {products?.map((item) => {
          return (
            <CartItemListItem key={item.product.id}>
              <ProductImageLink href={`/products/${item.product.id}`}>
                <ProductImage
                  src={item.product.image}
                  alt={item.product.name}
                  width={100}
                  height={100}
                  loading="lazy"
                />
              </ProductImageLink>
              <ProductDetails>
                <div>
                  <ProductName>{item.product.name}</ProductName>
                  <Description>{item.product.description}</Description>
                </div>
                <PriceAndQuantity>
                  <Price>${item.product.price}</Price>
                  <QuantityWrapper>
                    <StyledButton
                      disabled={item.count <= 1}
                      onClick={() =>
                        handleDecreaseCount(item.product.id, item.product.name)
                      }
                    >
                      <FaMinus />
                    </StyledButton>
                    <Quantity>{item.count}</Quantity>
                    <StyledButton
                      onClick={() =>
                        handleIncreaseCount(item.product.id, item.product.name)
                      }
                    >
                      <FaPlus />
                    </StyledButton>
                  </QuantityWrapper>
                </PriceAndQuantity>
              </ProductDetails>
              <CloseIcon
                onClick={() =>
                  handleRemoveProduct(item.product.id, item.product.name)
                }
              />
            </CartItemListItem>
          );
        })}
      </ul>
    </CartItemContainer>
  );
};

export default CartItem;
