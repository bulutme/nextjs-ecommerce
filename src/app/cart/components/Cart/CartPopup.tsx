import { ReactNode } from "react";
import Popup from "../../../../components/Popup";
import CartItem from "./CartItem";
import styled from "styled-components";
import Button from "../../../../components/Button";
import Link from "next/link";
import { useCart } from "@/context/CartContext";

type CartPopupProps = {
  children: ReactNode;
};

const Container = styled.div`
  position: relative;
  padding-top: 48px;
  border: 1px solid ${({ theme }) => theme.colors.border};
`;

const Title = styled.h4`
  position: fixed;
  top: 0;
  z-index: 10;
  padding: 12px;
  width: 100%;
  background: ${({ theme }) => theme.colors.white};
  border-bottom: 1px solid ${({ theme }) => theme.colors.border};
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1rem;
  padding: 12px;
  overflow-y: scroll;
  max-height: 360px;
`;

const ButtonContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  bottom: 0;
  position: sticky;
  background: ${({ theme }) => theme.colors.white};
  padding: 12px;
  border-top: 1px solid ${({ theme }) => theme.colors.border};
`;

const TotalRow = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 0.875rem;
  padding: 8px;
`;

const PriceTitle = styled.span`
  font-size: 0.875rem;
`;

const TotalPrice = styled.span`
  font-weight: ${({ theme }) => theme.fontWeight.semibold};
  font-size: 1.25rem;
  color: ${({ theme }) => theme.colors.primary};
`;

const CartPopup = ({ children }: CartPopupProps) => {
  const { cart, totalItemCount, totalPrice } = useCart();
  const isPopupVisible = totalItemCount > 0;

  const content =
    cart &&
    cart.map((item) => (
      <CartItem key={item.id} brand={item.brand} products={item.products} />
    ));

  return (
    <Popup
      trigger={children}
      content={
        isPopupVisible && (
          <Container>
            <Title>My Cart ({totalItemCount} Products)</Title>
            <Content>{content}</Content>
            <ButtonContainer>
              <TotalRow>
                <PriceTitle>Total</PriceTitle>
                <TotalPrice>${totalPrice}</TotalPrice>
              </TotalRow>
              <Link href="/cart">
                <Button
                  $fullwidth={true}
                  $variant="rounded"
                  size="medium"
                  color="primary"
                >
                  Go to Cart
                </Button>
              </Link>
            </ButtonContainer>
          </Container>
        )
      }
    />
  );
};

export default CartPopup;
