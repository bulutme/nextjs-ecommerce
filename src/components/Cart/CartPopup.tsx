import { ReactNode } from "react";
import Popup from "../Popup";
import CartItem from "./CartItem";
import styled from "styled-components";
import Button from "../Button";
import Link from "next/link";
import { useCart } from "@/context/CartContext";

type CartPopupProps = {
  children: ReactNode;
};

const Container = styled.div`
  position: relative;
  padding-top: 48px;
  border: 1px solid ${({ theme }) => theme.colors.border};
`;

const Title = styled.h4`
  position: fixed;
  top: 0;
  z-index: 10;
  padding: 12px;
  width: 100%;
  background: white;
  border-bottom: 1px solid ${({ theme }) => theme.colors.border};
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0.6rem;
  padding: 12px;
  overflow-y: scroll;
  max-height: 360px;
`;

const ButtonContainer = styled.div`
  width: 100%;
  bottom: 0;
  position: sticky;
  background: white;
  padding: 12px;
  border-top: 1px solid ${({ theme }) => theme.colors.border};
`;

const CartPopup = ({ children }: CartPopupProps) => {
  const { cart, totalItemCount } = useCart();
  const isPopupVisible = totalItemCount > 0;

  const content = cart.map((item, index) => (
    <CartItem key={index} brand={item.brand} products={item.products} />
  ));

  return (
    <Popup
      trigger={children}
      content={
        isPopupVisible && (
          <Container>
            <Title>My Cart ({totalItemCount} Products)</Title>
            <Content>{content}</Content>
            <ButtonContainer>
              <Link href="/cart">
                <Button
                  $fullwidth={true}
                  $variant="rounded"
                  size="small"
                  color="primary"
                >
                  Go to Cart
                </Button>
              </Link>
            </ButtonContainer>
          </Container>
        )
      }
    />
  );
};

export default CartPopup;
