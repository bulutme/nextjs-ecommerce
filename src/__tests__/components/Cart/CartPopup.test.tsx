import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { CartProvider } from "@/context/CartContext";
import { ThemeProvider } from "styled-components";
import { theme } from "@/styles/theme";
import CartPopup from "@/app/cart/components/Cart/CartPopup";
import { ToastProvider } from "@/context/ToastContext";
import CartItem from "@/app/cart/components/Cart/CartItem";

const mockProduct = {
  id: 1,
  name: "Test Product",
  image: "/test-image.jpg",
  description: "This is a test product",
  price: 19.99,
};

const mockCartItemProps = {
  brand: "Test Brand",
  products: [
    {
      product: mockProduct,
      count: 2,
    },
  ],
};

test("CartPopup renders correctly", () => {
  render(
    <ThemeProvider theme={theme}>
      <ToastProvider>
        <CartProvider>
          <CartPopup>content</CartPopup>
        </CartProvider>
      </ToastProvider>
    </ThemeProvider>
  );

  expect(screen.queryByText("My Cart")).toBeNull();
});

test("CartPopup snapshot test", () => {
  const { asFragment } = render(
    <ThemeProvider theme={theme}>
      <ToastProvider>
        <CartProvider>
          <CartPopup>content</CartPopup>
        </CartProvider>
      </ToastProvider>
    </ThemeProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("CartItem snapshot test inside CartPopup", () => {
  const { asFragment } = render(
    <ThemeProvider theme={theme}>
      <ToastProvider>
        <CartProvider>
          <CartPopup>
            <CartItem {...mockCartItemProps} />
          </CartPopup>
        </CartProvider>
      </ToastProvider>
    </ThemeProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("Go to Cart button is rendered and clickable", () => {
  render(
    <ThemeProvider theme={theme}>
      <ToastProvider>
        <CartProvider>
          <CartPopup>
            <button onClick={() => {}}>Open Popup</button>
            <button onClick={() => {}}>Go to Cart</button>
          </CartPopup>
        </CartProvider>
      </ToastProvider>
    </ThemeProvider>
  );

  const openPopupButton = screen.getByText("Open Popup");
  fireEvent.click(openPopupButton);

  const goToCartButton = screen.getByText("Go to Cart");

  expect(openPopupButton).toBeInTheDocument();
  expect(goToCartButton).toBeInTheDocument();

  fireEvent.click(goToCartButton);
});
