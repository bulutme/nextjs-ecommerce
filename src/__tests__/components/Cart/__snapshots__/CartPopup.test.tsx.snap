// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem snapshot test inside CartPopup 1`] = `
<DocumentFragment>
  <div
    class="sc-ec3ca1e3-0 gIMvgl"
  >
    <div
      class="sc-ec3ca1e3-1 hJnOck"
    >
      <div
        class="sc-dc5b0434-0 giFKjC"
      >
        <ul>
          <a
            href="#"
          >
            <p
              class="sc-dc5b0434-2 sc-dc5b0434-3 cpobqY kKrzEW"
            >
              Test Brand
            </p>
          </a>
          <li
            class="sc-dc5b0434-1 YMAYA"
          >
            <img
              alt="Test Product"
              class="sc-dc5b0434-4 APmUX"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="100"
              src="/_next/image?url=%2Ftest-image.jpg&w=256&q=75"
              srcset="/_next/image?url=%2Ftest-image.jpg&w=128&q=75 1x, /_next/image?url=%2Ftest-image.jpg&w=256&q=75 2x"
              style="color: transparent;"
              width="100"
            />
            <div
              class="sc-dc5b0434-5 bAPRmk"
            >
              <div>
                <p
                  class="sc-dc5b0434-2 cpobqY"
                >
                  Test Product
                </p>
                <p
                  class="sc-dc5b0434-6 evDWiE"
                >
                  This is a test product
                </p>
              </div>
              <div
                class="sc-dc5b0434-8 fNUzTm"
              >
                <p
                  class="sc-dc5b0434-10 huSRjE"
                >
                  $19.99
                </p>
                <div
                  class="sc-dc5b0434-7 jkUVuu"
                >
                  <button
                    class="sc-dc5b0434-12 kptlbj"
                    data-testid="decrease-button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 448 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                      />
                    </svg>
                  </button>
                  <span
                    class="sc-dc5b0434-9 blBBfb"
                  >
                    2
                  </span>
                  <button
                    class="sc-dc5b0434-12 kptlbj"
                    data-testid="increase-button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 448 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <svg
              class="sc-dc5b0434-11 kznnGQ"
              data-testid="remove-button"
              fill="currentColor"
              fill-rule="evenodd"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M799.855 166.312c.023.007.043.018.084.059l57.69 57.69c.041.041.052.06.059.084a.118.118 0 0 1 0 .069c-.007.023-.018.042-.059.083L569.926 512l287.703 287.703c.041.04.052.06.059.083a.118.118 0 0 1 0 .07c-.007.022-.018.042-.059.083l-57.69 57.69c-.041.041-.06.052-.084.059a.118.118 0 0 1-.069 0c-.023-.007-.042-.018-.083-.059L512 569.926 224.297 857.629c-.04.041-.06.052-.083.059a.118.118 0 0 1-.07 0c-.022-.007-.042-.018-.083-.059l-57.69-57.69c-.041-.041-.052-.06-.059-.084a.118.118 0 0 1 0-.069c.007-.023.018-.042.059-.083L454.073 512 166.371 224.297c-.041-.04-.052-.06-.059-.083a.118.118 0 0 1 0-.07c.007-.022.018-.042.059-.083l57.69-57.69c.041-.041.06-.052.084-.059a.118.118 0 0 1 .069 0c.023.007.042.018.083.059L512 454.073l287.703-287.702c.04-.041.06-.052.083-.059a.118.118 0 0 1 .07 0Z"
              />
            </svg>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="sc-ec3ca1e3-2 dNnrhF"
    />
  </div>
</DocumentFragment>
`;

exports[`CartPopup snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-ec3ca1e3-0 gIMvgl"
  >
    <div
      class="sc-ec3ca1e3-1 hJnOck"
    >
      content
    </div>
    <div
      class="sc-ec3ca1e3-2 dNnrhF"
    />
  </div>
</DocumentFragment>
`;
